cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -126,4 +127,26 @@ protected final void addSpark(String spark) {
     Predicate<WorkflowContext> predicate) {
     return configurer.condition(predicate);
   }
+
+  /**
+   * Adds a local dataset instance to the {@link Workflow}.
+   * See {@link co.cask.cdap.api.dataset.DatasetDefinition} for details.
+   *
+   * @param datasetName name of the dataset instance
+   * @param typeName name of the dataset type
+   * @param properties dataset instance properties
+   */
+  protected final void createLocalDataset(String datasetName, String typeName, DatasetProperties properties) {
```

Comment:
1. Yes. It should not matter if the given dataset name already exist in the namespace scope. Will update the documentation for this method.

2. For the `createLocalDataset` currently I added only two methods which take the `typeName` as String. We can support other versions which take the `Class`, but in case of  `createDataset` these versions also adds the dataset module which gets deployed during the application deployment. Is Workflow expected to deploy those dataset modules as well?

3. Corrected the wiki, it should be `As a developer of the Workflow, I want the ability to specify that the particular dataset used in the Workflow is local to it, so that Workflow system can create it at the start of every run and cleans it up after the run completes.`